code/modules/integrated_electronics/subtypes/lists.dm

```diff
@@ -335,18 +363,66 @@
 		if(list_index > input_list.len)
 			set_pin_data(IC_OUTPUT, i, null)
 		else
-			set_pin_data(IC_OUTPUT, i, input_list[list_index])
+			var/data = input_list[list_index]
+			set_pin_data(IC_OUTPUT, i, data)
+			set_pin_data(IC_OUTPUT, i + number_of_pins, input_list[data])
```

Comment:
it will glitch with duplicate entry.

Reply:
yeah, that's just a caveat of associative lists. doesn't really matter. the same thing happens in BYOND.

Reply:
IMO, better to leave list deconstructor as it is. But add special arglist circuit. I've thought a lot about it.Reverse convertor will be appreciated too..

Reply:
Alright